src/overlays/actors/ovl_En_Fz/z_en_fz.c

```diff
@@ -26,73 +70,812 @@ const ActorInit En_Fz_InitVars = {
+    f32 scale;
+    s32 life;
+
+    primColor.r = 155;
+    primColor.g = 255;
+    primColor.b = 255;
+    primColor.a = 255;
+    envColor.r = 200;
+    envColor.g = 200;
+    envColor.b = 200;
+    accel.z = 0.0f;
+    accel.y = -1.0f;
+    accel.x = 0.0f;
+
+    for (i = 0; i < numEffects; i++) {
+        scale = Math_Rand_CenteredFloat(0.3f) + 0.6f;
+        life = (s32)Math_Rand_CenteredFloat(5.0f) + 12;
+        pos.x = Math_Rand_CenteredFloat(unk_float) + vec->x;
+        pos.y = Math_Rand_ZeroFloat(unk_float) + vec->y;
+        pos.z = Math_Rand_CenteredFloat(unk_float) + vec->z;
+        vel.x = Math_Rand_CenteredFloat(10.0f);
+        vel.y = Math_Rand_ZeroFloat(10.0f) + 2.0f;
+        vel.z = Math_Rand_CenteredFloat(10.0f);
+        EffectSsEnIce_Spawn(globalCtx, &pos, scale, &vel, &accel, &primColor, &envColor, life);
+    }
+
+    func_80062CD4(globalCtx, vec);
+}
+
+void EnFz_SpawnIceSmokeHiddenState(EnFz* this) {
+}
+
+// Fully grown
+void EnFz_SpawnIceSmokeGrowingState(EnFz* this) {
+    Vec3f pos;
+    Vec3f velocity;
+    Vec3f accel;
+
+    if ((this->counter % 16) == 0) {
+        pos.x = Math_Rand_CenteredFloat(40.0f) + this->actor.posRot.pos.x;
+        pos.y = Math_Rand_CenteredFloat(40.0f) + this->actor.posRot.pos.y + 30.0f;
+        pos.z = Math_Rand_CenteredFloat(40.0f) + this->actor.posRot.pos.z;
+        accel.z = 0.0f;
+        accel.x = 0.0f;
+        velocity.z = 0.0f;
+        velocity.y = 0.0f;
+        velocity.x = 0.0f;
+        accel.y = 0.1f;
+        EnFz_SetIceSmokeNotFreezing(this, &pos, &velocity, &accel, Math_Rand_ZeroFloat(7.5f) + 15.0f);
+    }
+}
+
+// (2) Growing or Shrinking to/from hiding or (3) melting from fire
+void EnFz_SpawnIceSmokeActiveState(EnFz* this) {
+    Vec3f pos;
+    Vec3f velocity;
+    Vec3f accel;
+
+    if ((this->counter % 4) == 0) {
+        pos.x = Math_Rand_CenteredFloat(40.0f) + this->actor.posRot.pos.x;
+        pos.y = this->IceSmokeFreezingSpawnHeight;
+        pos.z = Math_Rand_CenteredFloat(40.0f) + this->actor.posRot.pos.z;
+        accel.z = 0.0f;
+        accel.x = 0.0f;
+        velocity.z = 0.0f;
+        velocity.y = 0.0f;
+        velocity.x = 0.0f;
+        accel.y = 0.1f;
+        EnFz_SetIceSmokeNotFreezing(this, &pos, &velocity, &accel, Math_Rand_ZeroFloat(7.5f) + 15.0f);
+    }
+}
+
+void EnFz_ApplyDamage(EnFz* this, GlobalContext* globalCtx) {
+    Vec3f vec;
+
+    if (this->isMoving && ((this->actor.bgCheckFlags & 8) ||
+                           (func_800339B8(&this->actor, globalCtx, 60.0f, this->actor.posRot.rot.y) == 0))) {
+        this->actor.bgCheckFlags &= ~8;
+        this->isMoving = false;
+        this->xzSpeed = 0.0f;
+        this->actor.speedXZ = 0.0f;
+    }
+
+    if (this->isFreezing) {
+        if ((this->actor.params < 0) && (this->collider1.base.atFlags & 2)) {
+            this->isMoving = false;
+            this->xzSpeed = 0.0f;
+            this->collider1.base.acFlags &= ~2;
+            this->actor.speedXZ = 0.0f;
```

Comment:
```suggestion
            this->actor.speedXZ =this->xzSpeed = 0.0f;
            this->collider1.base.acFlags &= ~2;
```
Assuming it matches. You may also want to use `speedXZ` instead of `xzSpeed` for consistency.

Reply:
Yup it matches! Just have to put it below collider though. Also consistency is good